src/test/java/com/squareup/kotlinpoet/KotlinPoetTest.kt

```diff
@@ -552,4 +545,50 @@ class KotlinPoetTest {
         |fun addA(s: String): String = s + "a"
         |""".trimMargin())
   }
+
+  @Test fun longParameterListWrapping() {
+    val source = FunSpec.builder("sum")
+        .addParameter(ParameterSpec.builder("a", Int::class).build())
+        .addParameter(ParameterSpec.builder("b", Int::class).build())
+        .addParameter(ParameterSpec.builder("c", Int::class).build())
+        .addParameter(ParameterSpec.builder("d", Int::class).build())
+        .addParameter(ParameterSpec.builder("e", Int::class).build())
+        .addParameter(ParameterSpec.builder("f", Int::class).build())
+        .addParameter(ParameterSpec.builder("g", Int::class).build())
+        .addStatement("return a + b + c")
+        .build()
+    assertThat(source.toString()).isEqualTo("""
+      |fun sum(
+      |    a: kotlin.Int,
+      |    b: kotlin.Int,
+      |    c: kotlin.Int,
+      |    d: kotlin.Int,
+      |    e: kotlin.Int,
+      |    f: kotlin.Int,
+      |    g: kotlin.Int
+      |) = a + b + c
+      |""".trimMargin())
+  }
+
+  @Test fun longLambdaParameterListWrapping() {
+    val source = FunSpec.builder("veryLongFunctionName")
+        .addParameter(ParameterSpec.builder(
+            "veryLongParameterName",
+            LambdaTypeName.get(
+                parameters = listOf(
+                    ParameterSpec.unnamed(Serializable::class),
+                    ParameterSpec.unnamed(Appendable::class),
+                    ParameterSpec.unnamed(Cloneable::class)),
+                returnType = Unit::class.asTypeName()))
+            .build())
+        .addParameter("i", Int::class)
+        .addStatement("return %T", Unit::class)
+        .build()
+    assertThat(source.toString()).isEqualTo("""
+      |fun veryLongFunctionName(
```

Comment:
Not sure what the style guide says about that, but having nested wrapping felt like an overkill. Hence, lambda types as function parameters are always single-line, even if they wrap.

Reply:
This behavior should also be consistent with the discussion on #256: we don't wrap the lambda parameters, but we do wrap the function parameters.